exporter/elasticsearchexporter/internal/objmodel/objmodel.go

```diff
@@ -424,7 +430,10 @@ func (doc *Document) iterJSONDedot(w *json.Visitor, otel bool) error {
 func StringValue(str string) Value { return Value{kind: KindString, str: str} }
 
 // IntValue creates a new value from an integer.
-func IntValue(i int64) Value { return Value{kind: KindInt, primitive: uint64(i)} }
+func IntValue(i int64) Value { return Value{kind: KindInt, i: i} }
+
+// UIntValue creates a new value from an unsigned integer.
+func UIntValue(i uint64) Value { return Value{kind: KindInt, ui: i} }
```

Comment:
I'd actually prefer a new kind for UInt to avoid a `v.ui != 0` check down the line. In fact, if there is a KindUint, both uint and int can share the same underlying primitive storage instead of having both `i` and `ui`.